feat: network checklist

arzola opened the pull request:
This PR address https://github.com/pressbooks/pressbooks-network-analytics/issues/316

Network managers will enjoy a new onboarding checklist that will help them to onboard users in a better way.

This would require to setup the following .env variables

```env
PB_CHECKLIST_BOOKING_URL=https://calendly.com/support-calendar
PB_CHECKLIST_NETWORK_CREATION_MONTHS_AGO="-6 months" #desired network age
PB_CHECKLIST_ONBOARDING_SURVEY=https://link-to-the-survey
```

In order to test this you need to modify your PB installation main site registration date to be within the timeframe specified in `PB_CHECKLIST_NETWORK_CREATION_MONTHS_AGO` meaning the site should be less than 6 months old if is older it won't display the checklist

### Test scenarios

1. All network managers should be able to modify the checklist, you can try different network managers
2. Values are preserved in the database
3. SSO items only are enabled if saml or cas plugins are enabled
4. Support items are only enabled if network analytics plugin is enabled.

arzola committed:
feat: network checklist

SteelWagstaff committed:
Update copy, styles, and display logic

SteelWagstaff commented on inc/admin/dashboard/class-networkdashboard.php:
```diff
@@ -55,4 +56,75 @@ protected function getTotalNumberOfBooks(): int {
+			],
+		];
+
+		// Check if either SSO plugin is activated
```
@arzola We can probably improve the logic here? DRY would be helpful, all that changes if CAS is active instead of SAML is the link URL ...

SteelWagstaff commented on inc/admin/dashboard/class-networkdashboard.php:
```diff
@@ -55,4 +56,75 @@ protected function getTotalNumberOfBooks(): int {
+			array_splice( $items, 3, 0, [ $sso_item ] );
+		}
+
+		// Check if Network Analytics plugin is activated
```
@arzola I added some conditional logic so that these menu items will only be displayed to users on networks that we host -- open source users won't see the invitation to join the private forum for our clients or the link to schedule a meeting with Amy :joy:

SteelWagstaff commented on templates/admin/dashboard/network.blade.php:
```diff
@@ -33,7 +33,55 @@ class="button button-primary"
+				<h2>{{ __( 'Congratulations', 'pressbooks' ) }}</h2>
+				<p>{{ __( 'You are ready to launch your Pressbooks network by inviting users to begin creating and publishing books!' ) }}</p>
+				<p>{{ __( 'Please help us understand and improve the network manager onboarding experience by completing a brief "Readiness to Launch" survey.', 'pressbooks' ) }}</p>
+				<a class="button button-primary" href="#">Take onboarding survey</a>
```
I made this a button so it's a more obvious CTA. We still need a survey to link here. Not sure when that will be complete and ready to share.

SteelWagstaff committed:
Update copy & styles for Congratulations message

arzola committed:
fix: store hook

arzola committed:
fix: save state change

arzola committed:
fix: unset composer changes

arzola commented:
@SteelWagstaff I improved the way we added the SSO checklist to avoid repetition and also removed a couple of unnecessary variables.

This should be working as expected, I also addressed the feature of only show this for networks that are less than 3 months old.

arzola committed:
style: solve lint issues

SteelWagstaff committed:
fix: minor copy changes to Congratulations message

SteelWagstaff committed:
fix: improve Congratulations message

arzola committed:
fix: store toggle

arzola committed:
fix: store reset and tests

arzola committed:
style: solve lint issues

richard015ar commented:
@arzola I am wondering if environment variables would be needed here instead of constants. It's because we need it more dynamic and possibly different for each environment, right?
I remember a similar discussion about https://github.com/pressbooks/pressbooks-vip/blob/b55646e1a4282f6144ca6929a5479c8c870a98ce/inc/class-subscription.php#L32

`PB_CHECKLIST_BOOKING_URL` and `PB_CHECKLIST_ONBOARDING_SURVEY` are going to be static, or we want change it better through infra level/env vars?

arzola commented:
@richard015ar The main reason to set env variables is to avoid spreading those values that could change in the future and also not relevant for OS users

SteelWagstaff committed:
fix: improve Congratulations block styles, linting

SteelWagstaff committed:
Merge branch 'feat/network-checklist' of https://github.com/pressbooks/pressbooks into feat/network-checklist

arzola committed:
fix: remove unused code

richard015ar commented on inc/admin/dashboard/class-networkdashboard.php:
```diff
@@ -55,4 +63,140 @@ protected function getTotalNumberOfBooks(): int {
+
+		if ( $network_creation_date ) {
+			$current_date = current_time( 'Y-m-d' ); // Use a non-timestamp format
+			$months_ago = strtotime( env( 'PB_CHECKLIST_NETWORK_CREATION_MONTHS_AGO' ) );
```
We should add a check here to ensure the env variable is defined, especially for open-source users.
Right now if the variable is not defined I got:
```Deprecated: strtotime(): Passing null to parameter #1 ($datetime) of type string is deprecated in /app/web/app/plugins/pressbooks/inc/admin/dashboard/class-networkdashboard.php on line 195 ```

Maybe adding this at the beginning?
```php
if ( ! env( 'PB_CHECKLIST_NETWORK_CREATION_MONTHS_AGO' )  ) {
    return false;
}
```

richard015ar commented:
Changes LGTM, I just left a comment for a tiny change.

arzola committed:
fix: return false if env not defined

richard015ar reviewed (approved):
LGTM! Thank you @arzola and @SteelWagstaff!

arzola committed:
chore: remove unwanted space
